django/template/base.py

```diff
@@ -187,12 +187,13 @@ def __init__(self, template_string, origin=None, name=None, engine=None):
         if engine is None:
             from .engine import Engine
             engine = Engine.get_default()
-        if engine.debug and origin is None:
```

Comment:
Given that origin is going to change further, I don't plan on documenting this change yet.